packages/app-store/office365calendar/lib/CalendarService.ts

```diff
@@ -689,17 +690,49 @@ export default class Office365CalendarService implements Calendar {
   async getMainTimeZone(): Promise<string> {
     try {
       const response = await this.fetcher(`${await this.getUserEndpoint()}/mailboxSettings/timeZone`);
-      const timezone = await handleErrorsJson<string>(response);
+      const timezoneResponse = await handleErrorsJson<string | { value: string }>(response);
 
-      if (!timezone || typeof timezone !== "string") {
+      let windowsTimezoneName: string;
+
+      if (typeof timezoneResponse === "object" && timezoneResponse !== null && "value" in timezoneResponse) {
+        windowsTimezoneName = timezoneResponse.value;
+        this.log.info("timezone found in outlook mailbox settings (new format)", {
+          windowsTimezoneName,
+        });
+      } else if (typeof timezoneResponse === "string") {
```

Comment:
So up until now we only considered strings - if `const timezone` was a string we returned it.

But now we try to 
```
const ianaTimezone = findIana(windowsTimezoneName);

```

Unless I am missing something we would be changing the behaviour in how we handle "string" timezones:

before: just return the original string
after: if you can convert it from windows IANA return that otherwise return the original string

and my guess is that we would return from windows IANA but are we okay that the first result in the array might not be same as original string?